clj-refactor.el

```diff
@@ -3992,18 +4000,23 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-change-function-
       (pop-to-buffer cljr--change-signature-buffer))))
 
 ;;;###autoload
-(defun cljr--inject-jack-in-dependencies ()
+(defun cljr--inject-jack-in-dependencies (cider-jack-in &rest args)
   "Inject the REPL dependencies of clj-refactor at `cider-jack-in'.
 If injecting the dependencies is not preferred set `cljr-inject-dependencies-at-jack-in' to nil."
-  (when (and cljr-inject-dependencies-at-jack-in
-             (boundp 'cider-jack-in-lein-plugins)
-             (boundp 'cider-jack-in-nrepl-middlewares))
-    (add-to-list 'cider-jack-in-lein-plugins `("refactor-nrepl" ,(cljr--version t)))
-    (add-to-list 'cider-jack-in-nrepl-middlewares "refactor-nrepl.middleware/wrap-refactor")))
+  (if (and (not (string-empty-p (cljr--project-dir)))
+           cljr-inject-dependencies-at-jack-in)
+      (let ((cider-jack-in-lein-plugins
+             (cons `("refactor-nrepl" ,(cljr--version t))
+                   cider-jack-in-lein-plugins))
+            (cider-jack-in-nrepl-middlewares
+             (cons "refactor-nrepl.middleware/wrap-refactor"
+                   cider-jack-in-nrepl-middlewares)))
+        (apply cider-jack-in args))
+    (apply cider-jack-in args)))
```

Comment:
why do you prefer `cons` over `add-to-list`?

Reply:
Because `add-to-list` would modify the list variable globally, unless I `let`-bound the variable to itself (which would be more complicated than just using `cons`).

Reply:
I see no harm in modifying these variables globally. Am I missing any edge cases here?

Reply:
If the variables were modified globally, then that would cause refactor-nrepl to be injected always, as opposed to just when you are inside a project. Avoiding that behavior is the whole point of this patch.